src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -47,54 +52,308 @@ export class PermissionsTableComponent implements OnInit {
   public entities: Entity[];
   public otherEntities: string[];
 
-  public constructor(private rolesService: RolesService) {
+  public rolesCheckbox;
+
+  public constructor(private rolesService: RolesService,
+                     private usersService: UsersService,
+                     private groupsService: GroupsService,
+                     private route: ActivatedRoute) {
   }
 
   public ngOnInit() {
     this.roles = ROLES[this.resourceType];
-    this.entities = this.getEntities(this.entityType);
-    this.otherEntities = this.getOtherEntities(this.entityType);
+    this.rolesCheckbox = {};
+    this.roles.forEach(r => {
+      this.rolesCheckbox[r] = false;
+    });
+    switch (this.resourceType) {
+      case 'organization':
```

Comment:
Adding new cases will provide that. But I wanted to implement it once CRUD operations for collections and views are implemented in UI.